moto/sagemaker/models.py

```diff
@@ -0,0 +1,364 @@
+        self.notebook_instances = {}
+        self.region_name = region_name
+
+    def reset(self):
+        region_name = self.region_name
+        self.__dict__ = {}
+        self.__init__(region_name)
+
+    def create_model(self, **kwargs):
+        model_obj = Model(
+            self.region_name,
+            kwargs.get("ModelName"),
+            kwargs.get("ExecutionRoleArn"),
+            kwargs.get("PrimaryContainer", {}),
+            kwargs.get("VpcConfig", {}),
+            kwargs.get("Containers", []),
+            kwargs.get("Tags", []),
+        )
+
+        self._models[kwargs.get("ModelName")] = model_obj
+        return model_obj.response_create
+
+    def describe_model(self, model_name=None):
+        for model in self._models.values():
+            # If a registry_id was supplied, ensure this repository matches
+            if model.ModelName != model_name:
+                continue
+            return model.response_object
+        message = "Could not find model '{}'.".format(
+            Model.arn_for_model_name(model_name, self.region_name)
+        )
+        raise RESTError(
+            error_type="ValidationException", message=message, template="error_json",
+        )
+
+    def list_models(self):
+        models = []
+        for model in self._models.values():
+            model_response = model.response_object.copy()
+            model_response.update(model.response_create)
+            models.append(model_response)
+        return {"Models": models}
+
+    def delete_model(self, model_name=None):
+        for model in self._models.values():
+            if model.ModelName == model_name:
+                self._models.pop(model.ModelName)
+                break
+        else:
+            raise MissingModel(model=model_name)
+
+    def create_notebook_instance(
+        self,
+        notebook_instance_name,
+        instance_type,
+        role_arn,
+        subnet_id=None,
+        security_group_ids=None,
+        kms_key_id=None,
+        tags=None,
+        lifecycle_config_name=None,
+        direct_internet_access="Enabled",
+        volume_size_in_gb=5,
+        accelerator_types=None,
+        default_code_repository=None,
+        additional_code_repositories=None,
+        root_access=None,
+    ):
+        self._validate_unique_notebook_instance_name(notebook_instance_name)
+
+        notebook_instance = FakeSagemakerNotebookInstance(
+            self.region_name,
+            notebook_instance_name,
+            instance_type,
+            role_arn,
+            subnet_id=subnet_id,
+            security_group_ids=security_group_ids,
+            kms_key_id=kms_key_id,
+            tags=tags,
+            lifecycle_config_name=lifecycle_config_name,
+            direct_internet_access=direct_internet_access
+            if direct_internet_access is not None
+            else "Enabled",
+            volume_size_in_gb=volume_size_in_gb if volume_size_in_gb is not None else 5,
```

Comment:
If I'm not mistaken, we can drop the if-else statements here.

To test this: extend `test_create_notebook_instance_minimal_params` to pass in `direct_internet_access=None, volume_size_in_gb=None`. The boto3-client should simply drop the arguments from the incoming request, and subsequently use the default function values from this method

Reply:
I think it's still necessary, since argument direct_internet_access could be _None_ (if explicitly passed in as such), so this ternary would be needed to default the value passed to FakeSagemakerNotebookInstance to "Enabled" when that's the case.

Reply:
Yup, you're right